### PR TITLE
MES-3107 autosave status

### PR DESCRIPTION
**Pull request by edwardwilson**
## Description and relevant Jira numbers
Simple change at add a new status into the Test enum
## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [ ] Reviewers selected in Github
- [x] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [x] Tested by QA
- [x] PO's approval

### CODE REVIEW

**Comment by szabikr** [comment]
I was just looking at these changes and couldn't really find where the `Autosave` status can come to the picture. To me autosave seems like a verb and not a status.

As my understanding goes we want to autosave the test result at that time when the test enters into the `WriteUp` status.

What would be the use of this new status?

**Comment by edwardwilson** [comment]
https://jira.dvsacloud.uk/browse/MES-2620

This is a state to indicate that the test has been autosaved, once the test gets to the write up state we will attempt an autosave. When successful the test state will be 'autosave'.

Once the writeup is complete then it will move to submitted; just as before.

**Comment by szabikr** [comment]
I would introduce a more complex object as a test status:
`{ status: 'Decided', hasBeenAutosaved: false }` -> `{ status: 'WriteUp', hasBeenAutosaved: true }`

File: src/modules/tests/test-status/test-status.model.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ export enum TestStatus {
   Started = 'Started',
   Decided = 'Decided',
   WriteUp = 'WriteUp',
+  Autosave = 'Autosave',
```
Comment (ammarhaiderbjss) [inline src/modules/tests/test-status/test-status.model.ts]: As discussed, can we please change to `Autosaved`?

**Review by ammarhaiderbjss — APPROVED** [review]

**Review by MarkPatt89 — APPROVED** [review]

**Review by rstepnowski — APPROVED** [review]

**Pull request merged**